vertx-config-vault/src/main/java/io/vertx/config/vault/client/SlimVaultClient.java

```diff
@@ -0,0 +1,337 @@
+    Integer port = configuration.getInteger("port", 8200);
+
+    Objects.requireNonNull(host, "The Vault host must be set");
+
+    client = WebClient.create(vertx, new WebClientOptions(configuration)
+      .setDefaultPort(port)
+    );
+
+    setToken(configuration.getString("token"));
+  }
+
+  /**
+   * Closes the client.
+   */
+  public void close() {
+    if (client != null) {
+      client.close();
+    }
+  }
+
+
+  /**
+   * Reads a secret from `path`.
+   *
+   * @param path            the path
+   * @param responseHandler the callback invoked with the result
+   */
+  public void read(String path, Handler<AsyncResult<Secret>> responseHandler) {
+    Objects.requireNonNull(responseHandler);
+
+    client.get("/v1/" + Objects.requireNonNull(path))
+      .putHeader(TOKEN_HEADER, Objects.requireNonNull(getToken(), "No token to access the vault"))
+      .send(response -> {
+        if (response.failed()) {
+          responseHandler.handle(VaultException.toFailure("Unable to access the Vault", response.cause()));
+          return;
+        }
+
+        HttpResponse<Buffer> result = response.result();
+        if (result.statusCode() != 200) {
+          responseHandler.handle(VaultException.toFailure(result.statusMessage(), result.statusCode(),
+            result.bodyAsString()));
+        } else {
+          Secret secret = result.bodyAsJson(Secret.class);
+          responseHandler.handle(Future.succeededFuture(secret));
+        }
+      });
+  }
+
+  /**
+   * Write a secret to `path`.
+   *
+   * @param path          the path
+   * @param resultHandler the callback invoked with the result
+   */
+  public void write(String path, JsonObject secrets, Handler<AsyncResult<Secret>> resultHandler) {
+    Objects.requireNonNull(resultHandler);
+    client.post("/v1/" + Objects.requireNonNull(path))
+      .putHeader(TOKEN_HEADER, Objects.requireNonNull(getToken(), "The token must be set"))
+      .sendJsonObject(Objects.requireNonNull(secrets, "The secret must be set"),
+        ar -> {
+          if (ar.failed()) {
+            resultHandler.handle(VaultException.toFailure("Unable to access the Vault", ar.cause()));
+            return;
+          }
+
+          HttpResponse<Buffer> response = ar.result();
+          if (response.statusCode() == 200 || response.statusCode() == 204) {
+            resultHandler.handle(Future.succeededFuture(response.bodyAsJson(Secret.class)));
+          } else {
+            resultHandler.handle(VaultException.toFailure(response.statusMessage(), response.statusCode(),
+              response.bodyAsString()));
+          }
+        });
+  }
+
+  /**
+   * Lists secrets from path (children).
+   *
+   * @param path          the path
+   * @param resultHandler the callback invoked with the result
+   */
+  public void list(String path, Handler<AsyncResult<List<String>>> resultHandler) {
+    String fullPath = path == null ? "list=true" : path + "?list=true";
```

Comment:
if `list=true` is a GET parameters, it should be preceded with `?` even if `path == null`